swag/blocks/cagnotte_blocks.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Union
+
+from swag.id import CagnotteId, UserId
+
+if TYPE_CHECKING:
+    from ..blockchain import SwagChain
+
+from attr import attrs, attrib
+
+from ..artefacts import CagnotteAccount
+
+from ..block import Block
+
+from ..errors import (
+    CagnotteDestructionForbidden,
+    CagnotteNameAlreadyExist,
+    NotCagnotteManager,
+)
+
+from ..currencies import Swag, Style
+
+
+@attrs(frozen=True, kw_only=True)
+class CagnotteCreation(Block):
+    cagnotte_id = attrib(type=CagnotteId, converter=CagnotteId)
+    name = attrib(type=str)
+    creator = attrib(type=UserId, converter=UserId)
+
+    def validate(self, db: SwagChain):
+        cagnotte_names = (cagnotte.name for cagnotte in db._accounts.cagnottes.values())
+        if self.cagnotte_id in db._accounts.cagnottes or self.name in cagnotte_names:
+            raise CagnotteNameAlreadyExist
+
+    def execute(self, db: SwagChain):
+        db._accounts[self.cagnotte_id] = CagnotteAccount(self.name, [self.creator])
+
+
+@attrs(frozen=True, kw_only=True)
+class CagnotteRenaming(Block):
+    cagnotte_id = attrib(type=CagnotteId, converter=CagnotteId)
+    new_name = attrib(type=str)
+
```

Comment:
Ooooh, right. I missed that one. But, actually, I think I copy-pasted code from the $wagBank, so either I missed it, either it wasn't there from the start 😓.

On de different note, I was thinking of removing every validate function so everything is inside the execute function. Why? Because part of the validation is in the execute function (cf the automatic exception in case you don't have enough money for a transaction), and it tends to duplicate some code (like `_accounts[]` accesses)